### DISCUSSION TITLE
Termination event times match current time?

### DISCUSSION THREAD

**Discussion by jcam** [aws/aws-node-termination-handler#606] [Q&A] [answered]
Termination events seem to always have a time matching the current time. Am I misconfigured?
An example below, note that the time of the log is 18:18:40, the 'will be interrupted is 18:18:38, which is 2 seconds ago, not 2 minutes in the future.
I am running in queue mode, with an eventbridge rule sending a message directly to an sqs queue, with the termination handler monitoring the queue. I am guessing that since it received a spot interruption event that I have the eventbridge properly configured.

2022/03/21 18:18:40 INF Adding new event to the event store event={"AutoScalingGroupName":"eks-spot-20220320215137259900000017","Description":"Spot Interruption event received. Instance i-043d45e1968bebec7 will be interrupted at 2022-03-21 18:18:38 +0000 UTC \n","EndTime":"0001-01-01T00:00:00Z","EventID":"spot-itn-event-38623039373536612d303063362d633862322d333566362d626538613263363639653330","InProgress":false,"InstanceID":"i-043d45e1968bebec7","IsManaged":true,"Kind":"SQS_TERMINATE","NodeLabels":null,"NodeName":"ip-10-89-12-166.ec2.internal","NodeProcessed":false,"Pods":null,"StartTime":"2022-03-21T18:18:38Z","State":""}

**Comment by LikithaVemulapalli** [comment, marked as answer]
Hello @jcam, we are returning the EventBridge event start time, in our case when instance received spot interruption. Current time is returned when we cannot parse the time in the specific format and we get a warning message that `Unable to parse time as RFC3339 from event, using current time instead.`, we cannot hardcode the time for spot interruption as it might take 2 or minutes to terminate the instance. But, yes the message needs to be modified to make it more clear, instead of mentioning as `will be interrupted`.
